File: snippets/csharp/System/NullReferenceException/Overview/Array3.cs
﻿using System;

public class Array3Example
{
    public static void Main()
    {
        // <Snippet10>
        int[] values = null;
        for (int ctr = 0; ctr <= 9; ctr++)
            values[ctr] = ctr * 2;

        foreach (int value in values)
            Console.WriteLine(value);

        // The example displays the following output:
        //    Unhandled Exception:
        //       System.NullReferenceException: Object reference not set to an instance of an object.
        //       at Array3Example.Main()
        // </Snippet10>
    }
}

File: snippets/fsharp/System/NullReferenceException/Overview/Array3.fs
module Array3

// <Snippet10>
let values: int[] = null
for i = 0 to 9 do
    values[i] <- i * 2

for value in values do
    printfn $"{value}"
// The example displays the following output:
//    Unhandled Exception:
//       System.NullReferenceException: Object reference not set to an instance of an object.
//       at <StartupCode$fs>.main()
// </Snippet10>